numpy/core/src/common/simd/avx2/arithmetic.h

```diff
@@ -118,6 +118,14 @@
 #endif // !NPY_HAVE_FMA3
 
 // Horizontal add: Calculates the sum of all vector elements.
+
+NPY_FINLINE npy_uint32 npyv_sum_u8(__m256i a)
```

Comment:
It should be named differently, what you do here is an extended sum. `npyv_sum_u32` produce overflow.

I suggest the following signatures:
```C
NPY_FINLINE npy_uint16 npyv_sumup_u8(npyv_u8 a);
NPY_FINLINE npy_uint32 npyv_sumup_u16(npyv_u16 a);
```

Reply:
The rename seems more reasonable. The [CI failure](https://travis-ci.com/github/numpy/numpy/jobs/473632623) of invalid intrinsics in VSX platform is a blocker for me, Do you have any suggestion?

Reply:
please see my comments below